src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt

```diff
@@ -108,15 +110,33 @@ class PayRatesService(
   private fun adjustOverlappingRates(
     req: PayRateRequest,
     expiredRates: MutableList<CourseActivityPayRate>,
-  ): PayRateRequest {
-    val expiredRatesForType = expiredRates.filter { exp -> exp.toRateType() == req.toRateType() }
-    return if (expiredRatesForType.isNotEmpty() && req.startDate!! <= expiredRatesForType.last().endDate) {
-      req.copy(startDate = expiredRates.last().endDate?.plusDays(1))
+  ): PayRateRequest? {
+    val lastExpiryForType = expiredRates.filter { exp -> exp.toRateType() == req.toRateType() }.maxOfOrNull { it.endDate!! }
+    return if (req.endDate != null && lastExpiryForType != null && req.endDate <= lastExpiryForType) {
+      // The requested rate doesn't make sense when compared with the last expiry on NOMIS - this must be corrupted data
+      // so ignore the request in favour of what's already on NOMIS
+      null
+    } else if (lastExpiryForType != null &&
+      req.isActive() &&
+      req.startDate!! <= lastExpiryForType
+    ) {
+      // The requested rate overlaps with the last expiry on NOMIS - adjust the start date to the day after the last expiry
+      req.copy(startDate = lastExpiryForType.plusDays(1))
     } else {
```

Comment:
The NOMIS and DPS history might not match due to data that has been corrupted while the pay rate sync. was unaware of DPS pay rate start dates.

This tightens up the processing to favour NOMIS history over DPS history - because we can't replace the NOMIS history as this is what people were really paid.

Specifically this change fixes 2 bugs:
* ignore DPS history entirely if it is covered by NOMIS history
* only adjust the start date for **active** DPS rates, not DPS history